diff --git a/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m b/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
index 4600e41..824ba9d 100644
--- a/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
+++ b/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
@@ -76,8 +76,10 @@ - (NSData *)dataFromHexString:(NSString *)string {
                   (NSString *) token:
                   (RCTPromiseResolveBlock) resolve :(RCTPromiseRejectBlock)reject) {
     @try {
-        NSData *data = [self dataFromHexString:token];
-        [Intercom setDeviceToken:data];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            NSData *data = [self dataFromHexString:token];
+            [Intercom setDeviceToken:data];
+        });
 
         NSLog(@"sendTokenToIntercom");
         resolve(@(YES));
